frontend/src/security/RegisterPage.tsx

```diff
@@ -9,7 +9,7 @@ function RegisterPage() {
     const [username, setUsername] = useState<string>("")
     const [rawPassword, setRawPassword] = useState<string>("")
     const [registrationErrorMessage, setRegistrationErrorMessage] = useState({
-        rawPassword: undefined,
+        rawPassword: [],
         username: undefined,
```

Comment:
I would here create new variable for initial errorMessage state, and I will also use empty string instead of undefined, and it will be Falsy in you check .